lib/functions/transfer.js

```diff
@@ -60,16 +60,28 @@ class Transfer extends AsyncGeneratorFunction {
                 controller.notify(TransferEvents.TRANSFER, this.name, transferPart);
 
                 const targetUrl = Array.isArray(transferPart.targetUrls) && (transferPart.targetUrls.length === 1) && transferPart.targetUrls[0];
-                const contentRanges = transferPart.contentRange.subranges(); 
+                const contentRanges = transferPart.contentRange.subranges();
                 if (contentRanges.length !== 1) {
                     throw Error(`'contentRanges' must be a single range: ${contentRanges}`);
                 }
+
                 const contentRange = contentRanges[0];
                 if (contentRange.length > MAX_MEMORY_BUFFER) {
                     throw Error(`'contentRange.length' too large, not supported yet: ${transferPart.source.url}`);
                 }
+
+                console.log(`Transferred content range: ${contentRange}`);
                 if (isFileProtocol(transferPart.source.url) && targetUrl) {
+                    console.log("Source has protocol 'file'");
+
                     const buf = await this.randomFileAccess.read(transferPart.source.url, contentRange);
+
+                    if (buf && buf.length) {
+                        console.log(`Read file ${transferPart.source.url} for content range low ${contentRange.low}, to high ${contentRange.high}, read ${buf.length} bytes`);
+                    } else {
```

Comment:
please confirm contentRange won't be `undefined`

Reply:
done